File: ventas_app/lib/features/telas/domain/entities/tela.dart
class Tela {
  final int id;
  final String nombre;
  final double? precxmen;
  final double? precxmay;
  final double? precxrollo;
  final double? precxcompra;
  final int? idusuarios;

  Tela(
      {this.precxmen,
      this.precxmay,
      this.precxrollo,
      this.precxcompra,
      this.idusuarios,
      required this.id,
      required this.nombre});

  // Constructor para crear una tela vacía
  factory Tela.empty() {
    return Tela(
      id: 0,
      nombre: '',
      precxmen: 0,
      precxmay: 0,
      precxrollo: 0,
      precxcompra: 0,
      idusuarios: 0,
    );
  }
}

File: ventasBackend/models/tela.js
/* const sequelize = require("../database/database");
const { DataTypes } = require('sequelize');
const Usuario = require('./usuario');

const Tela = sequelize.define('telas', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    nombre: {
        type: DataTypes.STRING,
    },
    precxmen: {
        type: DataTypes.FLOAT,
    },
    precxmay: {
        type: DataTypes.FLOAT,        
    },    
    precxrollo: {
        type: DataTypes.FLOAT,        
    },
    precxcompra: {
        type: DataTypes.FLOAT,        
    },
    estado: {
        type: DataTypes.BOOLEAN,
        defaultValue: true,
    },
    idusuarios: {
        type: DataTypes.INTEGER,
        references: {
            model: Usuario, 
            key: 'id'
        }
    }
}, {
    timestamps: false, 
});

Tela.belongsTo(Usuario, { foreignKey: 'idusuarios' }); 

module.exports = Tela;

 */